[Python] Missing arguments not flagged when using keyword arguments

For this program:
```python
def foo(w: int, x: int, y: int = 3) -> int:
    return x + y

foo(3)
```
ESBMC correctly flags the missing required arguments:
```bash
$ esbmc ex.py 
ESBMC version 7.11.0 64-bit x86_64 linux
Target: 64-bit little-endian x86_64-unknown-linux with esbmclibc
Parsing ex.py

Type checking warning:
ex.py:4: error: Missing positional argument "x" in call to "foo"  [call-arg]
Found 1 error in 1 file (checked 1 source file)

WARNING: foo() missing required positional argument: 'x'

* file: ex.py
* line 4
* function: foo
* column: 0 
Converting
Generating GOTO Program
GOTO program creation time: 0.546s
GOTO program processing time: 0.003s
Starting Bounded Model Checking
Symex completed in: 0.000s (3 assignments)
Caching time: 0.000s (removed 0 assertions)
Slicing time: 0.000s (removed 2 assignments)
Generated 1 VCC(s), 1 remaining after simplification (1 assignments)
No solver specified; defaulting to Boolector
Encoding remaining VCC(s) using bit-vector/floating-point arithmetic
Encoding to solver time: 0.000s
Solving with solver Boolector 3.2.3
Runtime decision procedure: 0.001s
Building error trace

[Counterexample]


State 1  thread 0
----------------------------------------------------
Violated property:
  function call: not enough arguments


VERIFICATION FAILED
```
However, if we use keyword arguments instead:
```python
def foo(w: int, x: int, y: int = 3) -> int:
    return x + y

foo(w=3)
```
verification unexpectedly succeeds:
```bash
$ esbmc ex.py 
ESBMC version 7.11.0 64-bit x86_64 linux
Target: 64-bit little-endian x86_64-unknown-linux with esbmclibc
Parsing ex.py

Type checking warning:
ex.py:4: error: Missing positional argument "x" in call to "foo"  [call-arg]
Found 1 error in 1 file (checked 1 source file)

WARNING: foo() missing required positional argument: 'x'

* file: ex.py
* line 4
* function: foo
* column: 0 
Converting
Generating GOTO Program
GOTO program creation time: 0.541s
GOTO program processing time: 0.003s
Starting Bounded Model Checking
Symex completed in: 0.000s (4 assignments)
Caching time: 0.000s (removed 0 assertions)
Slicing time: 0.000s (removed 4 assignments)
Generated 0 VCC(s), 0 remaining after simplification (0 assignments)
BMC program time: 0.000s

VERIFICATION SUCCESSFUL
```

The feature added in #3022 does not seem to apply to methods. For example, for:
```python
class Foo:
    def __init__(self) -> None:
        pass

    def foo(self, x: int, y: int) -> int:
        return x + y

f = Foo()
f.foo(x=3)
```
verification succeeds despite the missing argument (`y`):
```bash
$ esbmc ex.py
ESBMC version 7.11.0 64-bit x86_64 linux
Target: 64-bit little-endian x86_64-unknown-linux with esbmclibc
Parsing ex.py

Type checking warning:
ex.py:9: error: Missing positional argument "y" in call to "foo" of "Foo"  [call-arg]
Found 1 error in 1 file (checked 1 source file)

Converting
Generating GOTO Program
GOTO program creation time: 0.560s
GOTO program processing time: 0.002s
Starting Bounded Model Checking
Symex completed in: 0.000s (5 assignments)
Caching time: 0.000s (removed 0 assertions)
Slicing time: 0.000s (removed 5 assignments)
Generated 0 VCC(s), 0 remaining after simplification (0 assignments)
BMC program time: 0.001s

VERIFICATION SUCCESSFUL
```
Is it possible to extend the feature to capture missing arguments for methods as well? Let me know if you prefer that I file a separate issue instead.

--- a/regression/python/github_3010_4_fail/main.py
+++ b/regression/python/github_3010_4_fail/main.py
@@ -1,0 +1,9 @@
+class Foo:
+    def __init__(self) -> None:
+        pass
+
+    def foo(self, x: int, y: int) -> int:
+        return x + y
+
+f = Foo()
+f.foo(x=3)

--- a/regression/python/github_3010_4_fail/test.desc
+++ b/regression/python/github_3010_4_fail/test.desc
@@ -1,0 +1,4 @@
+CORE
+main.py
+
+^TypeError: foo\(\) missing 1 required positional argument: \'y\'$

--- a/src/python-frontend/preprocessor.py
+++ b/src/python-frontend/preprocessor.py
@@ -1059,15 +1059,30 @@ class Preprocessor(ast.NodeTransformer):
                 else:
                     node.args[1] = ast.Constant(value=False)
 
-        # if not a function or preprocessor doesn't have function definition return
-        if not isinstance(node.func,ast.Name) or node.func.id not in self.functionParams:
+        # Determine if this is a method call or function call
+        functionName = None
+        expectedArgs = None
+
+        if isinstance(node.func, ast.Attribute):
+            # Handle method calls (e.g., obj.method())
+            method_name = node.func.attr
+            if method_name in self.functionParams:
+                functionName = method_name
+                # For method calls, skip 'self' parameter (first parameter)
+                expectedArgs = self.functionParams[method_name][1:]
+        elif isinstance(node.func, ast.Name):
+            # Handle regular function calls
+            if node.func.id in self.functionParams:
+                functionName = node.func.id
+                expectedArgs = self.functionParams[functionName]
+
+        # If not a tracked function/method, just visit and return
+        if functionName is None or expectedArgs is None:
             self.generic_visit(node)
             return node
 
-        functionName = node.func.id
-        expectedArgs = self.functionParams[functionName]
-        keywords = {}
         # add keyword arguments to function call
+        keywords = {}
         for i in node.keywords:
             if i.arg in keywords:
                 raise SyntaxError(f"Keyword argument repeated:{i.arg}",(self.module_name,i.lineno,i.col_offset,""))

